specs/NotifyDropTargetAction.md

```diff
@@ -0,0 +1,313 @@
+<!-- TEMPLATE
+  The purpose of this spec is to describe a new WebView2 feature and its APIs.
+
+  There are two audiences for the spec. The first are people
+  that want to evaluate and give feedback on the API, as part of
+  the submission process. When it's complete
+  it will be incorporated into the public documentation at
+  docs.microsoft.com (https://docs.microsoft.com/en-us/microsoft-edge/webview2/).
+  Hopefully we'll be able to copy it mostly verbatim.
+  So the second audience is everyone that reads there to learn how
+  and why to use this API. 
+-->
+
+
+# Background
+These APIs will allow users to drop things such as images, text, and links into the WebView as part of a drag/drop operation.
+The reason that we need separate APIs for this with composition hosting is because we don't have an HWND to call RegisterDragDrop on.
+The hosting app needs to call RegisterDragDrop on the HWND that contains the WebView and implement IDropTarget so it can forward the
+calls (IDropTarget::DragEnter, DragMove, DragLeave, and Drop) to the WebView. Other UI frameworks have their own ways to register.
+For example, Xaml require setting event handler for DragEnter, DragOver, DragLeave, and Drop on the corresponding UIElement.
+
+Additionally, for win32, the hosting app also needs to call into IDropTargetHelper before forwarding those calls to us as documented here:
+https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nn-shobjidl_core-idroptargethelper
+
+For API reviewers, We want a unified API surface between COM and WinRT that works in both UWP and Win32.
+
+We could:
+1. Have one API focusing on Win32 types and require the end dev to convert from UWP types to Win32 (which we've done).
```

Comment:
Using UWP data package type but not using UWP keyboard state is inconsistent. WinUI will also invent something new (lifted version of UWP types).  Lets just make it all use Windows.ApplicationModel types in the WinRT API and Win32 types in our COM API.

Reply:
So basically do 3.) and then convert the types internally.